### ISSUE TITLE
Make install_version (or other install_ function) failures more verbose

### ISSUE THREAD

**Issue by colearendt** [r-lib/remotes#425]
This is an example of a failure that could do with a bit more logging:

```
devtools::install_version("A3", version = "0.9.1")
#> Error in package_find_repo(package, repos) : couldn't find package 'A3'
```

It makes debugging challenging because there is no information around what was tried, what URLs were attempted, etc. More information around why this failure is happening would be helpful. I.e.:

- the latest version of the package, if that's easily accessible
- the URL that is being reached for
- the response code

**Comment by colearendt** [comment]
Insightful tip: `options(internet.info = 0)` adds verbose logging for successes, at least.

**Comment by jimhester** [comment]
There isn't much we can do, the code that is failing is in base R.

Having the results of `traceback()` would help figure out what is failing though.

**Comment by jimhester** [comment]
Also could you provide a reproducible example, the example you give works fine with the default CRAN repositories.

**Comment by colearendt** [comment]
Oops. Yeah, sorry about that.

``` r
options("repos" = c("RSPM" = "https://demo.rstudiopm.com/internal/latest"))
devtools::install_version("A3", version = "0.9.1")
#> Error in package_find_repo(package, repos): couldn't find package 'A3'
```

<sup>Created on 2019-08-17 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by jimhester** [comment]
The A3 package isn't in the RSPM archive?

``` r
rspm_archive <- readRDS(gzcon(url(sprintf("%s/src/contrib/Meta/archive.rds", "https://demo.rstudiopm.com/internal/latest"), "rb")))
names(rspm_archive)
#> [1] "connectApiUtils" "dbplot"          "deepviz"         "ferris"         
#> [5] "rstudiointernal" "tidypredict"
```

<sup>Created on 2019-08-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

This seems to be working as intended, if you include the CRAN repo as well the package is found. It also gives you a message saying which repository the code is trying when there is more than one listed.

``` r
remotes:::package_find_repo("A3", c(RSPM = "https://demo.rstudiopm.com/internal/latest", CRAN = "https://cran.r-project.org"))
#> Trying https://demo.rstudiopm.com/internal/latest
#> Trying https://cran.r-project.org
#>                     size isdir mode               mtime
#> A3/A3_0.9.1.tar.gz 45252 FALSE  664 2013-02-07 04:00:29
#> A3/A3_0.9.2.tar.gz 45907 FALSE  664 2013-03-26 14:58:40
#>                                  ctime               atime  uid  gid
#> A3/A3_0.9.1.tar.gz 2019-08-18 00:03:09 2019-08-18 01:05:48 1001 1001
#> A3/A3_0.9.2.tar.gz 2019-08-18 00:03:09 2019-08-18 01:05:48 1010 1001
#>                     uname    grname                       repo
#> A3/A3_0.9.1.tar.gz hornik cranadmin https://cran.r-project.org
#> A3/A3_0.9.2.tar.gz ligges cranadmin https://cran.r-project.org
```

<sup>Created on 2019-08-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by colearendt** [comment]
Right, the A3 package is not in that repository 😆 Sorry. It's just a simple way to reproduce the behavior that doesn't give me any information around what was tried / why it failed. 

I'm not sure what debugging output would be possible, but the request here is for more of the logging to be available from the `install_version()` output. I.e. more than just "Sorry, I couldn't find it"

**Comment by JosiahParry** [comment]
I'm running to this same issue with `remotes::install_version()` on a local machine running R 3.6.1.

Running `remotes:::package_find_repo("dplyr", c("CRAN" = "https://cran.rstudio.com"))` returns the same error `couldn't find package 'dplyr'`. 

I've tried running the internal lines

```
con <- gzcon(url(sprintf("%s/src/contrib/Meta/archive.rds", 
                           c("CRAN" = "https://cran.rstudio.com")), "rb"))
readRDS(con)
```

But upon reading the RDS the session is terminated. I've then downloaded the `archive.rds` manually and tried reading that in which returned the following error message:

```
> load("archive.rds")
Error in load("archive.rds") : 
  bad restore file magic number (file may be corrupted) -- no data loaded
In addition: Warning message:
file ‘archive.rds’ has magic number 'X'
  Use of save versions prior to 2 is deprecated 
```
It seems at this point that the archive file is corrupted. 

But, oddly enough, if I run `remotes::install_version("dplyr", "0.7.0", repos = "https://cran.rstudio.com/")` on the Colorado RSP instance with both R v 3.5.3 and 3.6.2 it downloads successfully.
